### PR TITLE
improv: update metrics interface to resemble other core utils

### PR DESCRIPTION
**Pull request by cakepietoast**
**Issue #, if available:** #33 

## Description of changes:

> Update Metrics to accept a "service" parameter to match Logger/Tracer. This parameter will be used as the metric namespace. Deprecated MetricManager.add_namespace env var.

> BREAKING: Remove POWERTOOLS_METRICS_NAMESPACE, in favour of using the same environment variable as Logger/Tracer classes (POWERTOOLS_SERVICE_NAME).

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] Update tests
* [x] Update docs
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/376ec0a2ac0d2a40e0af5717bef42ff84ca0d1b9/.github/semantic.yml#L2)

### CODE REVIEW

File: docs/content/core/metrics.mdx [hunk 1]
```diff
@@ -27,16 +27,22 @@ Resources:
...
 ```python:title=app.py
 from aws_lambda_powertools.metrics import Metrics, MetricUnit
 
-metrics = Metrics()
-# metrics.add_namespace("ServerlessAirline") # optionally if you set via env var
+# POWERTOOLS_SERVICE_NAME defined
+metrics = Metrics() # highlight-line 
+
+# Explicit definition
+Logger(service="ServerlessAirline")  # sets namespace to "ServerlessAirline"
```
Comment (heitorlessa) [inline docs/content/core/metrics.mdx:43]:
```suggestion
Metrics(service="ServerlessAirline")  # sets namespace to "ServerlessAirline"
```

File: docs/content/index.mdx [hunk 2]
```diff
@@ -36,12 +36,11 @@ _`*` Core utilities are Tracer, Logger and Metrics. Optional utilities may vary
 
 Environment variable | Description | Utility
 ------------------------------------------------- | --------------------------------------------------------------------------------- | --------------------------------------------------------------------------------- | -------------------------------------------------
-**POWERTOOLS_SERVICE_NAME** | Sets service name used for tracing namespace, metrics dimensions and structured logging | all
+**POWERTOOLS_SERVICE_NAME** | Sets service name used for tracing namespace, metrics namespace and structured logging | all
```
Comment (heitorlessa) [inline docs/content/index.mdx:39]: Thanks for correcting the dimensions wording too! That's what I had in the first implementation as add_dimension was optional

**Comment by heitorlessa** [comment]
Are you still planning to add the `service` as a dimension? or will that be in a separate PR?

Could you also change docstring in line 89 in metrics.py:

```
        **Lambda function using tracer and metrics decorators**

            metrics = Metrics()
```

Thanks again, Tom!!

**Comment by cakepietoast** [comment]
> Are you still planning to add the `service` as a dimension? or will that be in a separate PR?

I'm in two minds about adding a dimension with the service name. I'll create a separate issue/PR for that to discuss.

> Could you also change docstring in line 89 in metrics.py:
> 
> ```
>         **Lambda function using tracer and metrics decorators**
> 
>             metrics = Metrics()
> ```
Done!
> 
> Thanks again, Tom!!

**Pull request merged**